### PR TITLE
action logging

### PR DESCRIPTION
**Pull request by mattuntergassmair**

### CODE REVIEW

File: src/actions/interface.jl [hunk 1]
```diff
@@ -6,5 +6,35 @@ entity based on the action on the given roadway.
...
+Mapping from actions to vehicle ids. The main use case is for keeping track of the action history
+in the same way as the scene history in the `simulate!` function.
+
+Initialize as
+
+    ActionMapping(a, id)
+
+where `a` is an action which can be used to propagate entities. `id` is the entity identifier.
+"""
+struct ActionMapping{A,I}
```
Comment (MaximeBouton) [inline src/actions/interface.jl]: is `ActionMapping` the best name for this? 
When I see mapping, I think dictionaries or functions. 
Maybe `EntityAction` or `AgentAction`?

File: src/actions/interface.jl [hunk 2]
```diff
@@ -6,5 +6,35 @@ entity based on the action on the given roadway.
...
+        am = frame.entities[am_index]
+        if am.id == id
+            return am_index
+        end
+    end
+    return nothing
+end
+function Records.id2index(frame::Frame{ActionMapping{A,I}}, id::I) where {A,I}
+    entity_index = findfirst(id, frame)
+    entity_index === nothing && throw(BoundsError(frame, id))
```
Comment (MaximeBouton) [inline src/actions/interface.jl]: is that line correct? 
I don't understand the syntax, it should behave as
```julia 
if entity_index === nothing
    throw(BoundsError(frame, id))
end
```
Are the two syntax equivalent? 

I like 
```julia
entity_index === nothing ? throw(BoundsError(frame, id)) : nothing 
```` 
if you are looking for a one liner.

File: src/deprecated.jl [hunk 3]
```diff
@@ -2,3 +2,40 @@
 @deprecate get_vel_t velf(state).t
 
 @deprecate propagate(veh, state, roadway, Δt) propagate(veh, state, a, roadway, Δt)
+
```
Comment (MaximeBouton) [inline src/deprecated.jl:5]: Can you add the `rec` version of `simulate!` in there two?

Comment (MaximeBouton) [reply src/deprecated.jl:5]: just add a deprecation warning to the `rec` version but leave them

**Comment by MaximeBouton** [comment]
Thanks for doing this! 🎉

File: src/deprecated.jl (hunk 3 above)
Comment (mattuntergassmair) [reply src/deprecated.jl:5]: Deprecated all versions of `simulate!` using `rec` for which we currently don't have an alternative implementation.

**Pull request merged**